app/models/order.rb

```diff
@@ -24,4 +24,12 @@ def save_purchaser_name
     self.purchaser_name ||= user && user.username
   end
 
+  def self.parse_params(params)
```

Comment:
This is a weird method name to me. Maybe something like sort_by_status instead? It's unclear from the name what the method does.